### PR TITLE
[docs] General updates

### PR DESCRIPTION
**Pull request by stevhliu**
Implements some of the easier improvements that we discussed:

* updates the supported pipelines table and move it from the `index` to the pipelines [overview](https://huggingface.co/docs/diffusers/api/pipelines/overview)
* removes the duplicate `DiffusionPipeline` API page [here](https://huggingface.co/docs/diffusers/api/diffusion_pipeline) because it is already documented in the pipelines overview
* adds a brief section about caching and how to use if offline
* updates the [community pipelines](https://huggingface.co/docs/diffusers/using-diffusers/custom_pipeline_examples) doc with some more interesting examples that work (open to showcasing other community pipelines here)

### CODE REVIEW

File: docs/source/en/index.md [hunk 1]
```diff
@@ -45,54 +45,4 @@ The library has three main components:
...
-| [versatile_diffusion](./api/pipelines/versatile_diffusion) | [Versatile Diffusion: Text, Images and Variations All in One Diffusion Model](https://arxiv.org/abs/2211.08332) | Text-to-Image Generation |
-| [versatile_diffusion](./api/pipelines/versatile_diffusion) | [Versatile Diffusion: Text, Images and Variations All in One Diffusion Model](https://arxiv.org/abs/2211.08332) | Image Variations Generation |
-| [versatile_diffusion](./api/pipelines/versatile_diffusion) | [Versatile Diffusion: Text, Images and Variations All in One Diffusion Model](https://arxiv.org/abs/2211.08332) | Dual Image and Text Guided Generation |
-| [vq_diffusion](./api/pipelines/vq_diffusion) | [Vector Quantized Diffusion Model for Text-to-Image Synthesis](https://arxiv.org/abs/2111.14822) | Text-to-Image Generation |
-| [stable_diffusion_ldm3d](./api/pipelines/stable_diffusion/ldm3d_diffusion) | [LDM3D: Latent Diffusion Model for 3D](https://arxiv.org/abs/2305.10853) | Text to Image and Depth Generation |
+</div>
```
Comment (patrickvonplaten) [inline docs/source/en/index.md:48]: It would be nice to link to a table of all existing pipelines here though and try to enforce keeping this table up to date

File: docs/source/en/api/diffusion_pipeline.md [hunk 2]
```diff
@@ -1,36 +0,0 @@
-<!--Copyright 2023 The HuggingFace Team. All rights reserved.
```
Comment (patrickvonplaten) [inline docs/source/en/api/diffusion_pipeline.md:1]: Hmm where is the diffusion pipeline now shown instead? The `DiffusionPipeline` is a very important object no?

File: docs/source/en/api/pipelines/overview.md [hunk 3]
```diff
@@ -12,16 +12,70 @@ specific language governing permissions and limitations under the License.
...
 <Tip warning={true}>
 
-Pipelines do not offer any training functionality. You'll notice PyTorch's autograd is disabled by decorating the [`~DiffusionPipeline.__call__`] method with a [`torch.no_grad`](https://pytorch.org/docs/stable/generated/torch.no_grad.html) decorator because pipelines should not be used for training. If you're interested in training, please take a look at the [Training](../traininig/overview) guides instead!
+Pipelines do not offer any training functionality. You'll notice PyTorch's autograd is disabled by decorating the [`~DiffusionPipeline.__call__`] method with a [`torch.no_grad`](https://pytorch.org/docs/stable/generated/torch.no_grad.html) decorator because pipelines should not be used for training. If you're interested in training, please take a look at the [Training](../../training/overview) guides instead!
 
 </Tip>
 
+The table below lists all the pipelines currently available in 🤗 Diffusers and the tasks they support. Click on a pipeline to view its abstract and published paper.
+
+| Pipeline | Tasks |
```
Comment (patrickvonplaten) [inline docs/source/en/api/pipelines/overview.md:27]: Nice that works for me!

File: docs/source/en/api/pipelines/overview.md [hunk 4]
```diff
@@ -12,16 +12,70 @@ specific language governing permissions and limitations under the License.
...
+| [ParaDiGMS](paradigms) | text2image |
+| [Pix2Pix Zero](pix2pix_zero) | image editing |
+| [PNDM](pndm) | unconditional image generation |
+| [RePaint](repaint) | inpainting |
+| [ScoreSdeVe](score_sde_ve) | unconditional image generation |
+| [Self-Attention Guidance](self_attention_guidance) | text2image |
+| [Semantic Guidance](semantic_stable_diffusion) | text2image |
+| [Shap-E](shap_e) | text-to-3D, image-to-3D |
+| [Spectrogram Diffusion](spectrogram_diffusion) |  |
+| [StableDiffusion](stable_diffusion/overview) | text2image, image2image, depth2image, inpainting, image variation, latent upscaler, super-resolution |
```
Comment (sayakpaul) [inline docs/source/en/api/pipelines/overview.md:63]: Maybe separate both words? "Stable Diffusion" instead of "StableDiffusion"?

File: docs/source/en/api/pipelines/overview.md [hunk 5]
```diff
@@ -12,16 +12,70 @@ specific language governing permissions and limitations under the License.
...
+| [StableDiffusionModelEditing](model_editing) | model editing |
```
Comment (sayakpaul) [inline docs/source/en/api/pipelines/overview.md:64]: Same as above.

File: docs/source/en/api/pipelines/overview.md [hunk 6]
```diff
@@ -12,16 +12,70 @@ specific language governing permissions and limitations under the License.
...
+| [ControlNet with Stable Diffusion XL](controlnet_sdxl) | text2image |
+| [Cycle Diffusion](cycle_diffusion) | image2image |
+| [Dance Diffusion](dance_diffusion) | unconditional audio generation |
+| [DDIM](ddim) | unconditional image generation |
+| [DDPM](ddpm) | unconditional image generation |
+| [DeepFloyd IF](deepfloyd_if) | text2image, image2image, inpainting, super-resolution |
+| [DiffEdit](diffedit) | inpainting |
+| [DiT](dit) | text2image |
+| [GLIGEN](gligen) | text2image |
+| [InstructPix2Pix](pix2pix) | image editing |
```
Comment (sayakpaul) [inline docs/source/en/api/pipelines/overview.md:46]: Seems like we're missing DeepFloyd's IF.

File: docs/source/en/installation.md [hunk 7]
```diff
@@ -123,17 +119,29 @@ git pull
...
 This usage data helps us debug issues and prioritize new features.
-Telemetry is only sent when loading models and pipelines from the HuggingFace Hub,
-and is not collected during local usage.
+Telemetry is only sent when loading models and pipelines from the Hub,
+and it is not collected if you're loading local files.
 
-We understand that not everyone wants to share additional information, and we respect your privacy,
-so you can disable telemetry collection by setting the `DISABLE_TELEMETRY` environment variable from your terminal:
+We understand that not everyone wants to share additional information,and we respect your privacy.
+You can disable telemetry collection by setting the `DISABLE_TELEMETRY` environment variable from your terminal:
```
Comment (sayakpaul) [inline docs/source/en/installation.md:144]: Nice rework!

**Review by sayakpaul — APPROVED** [review]
Thanks for changes!

File: docs/source/en/index.md (hunk 1 above)
Comment (stevhliu) [reply docs/source/en/index.md:48]: Added a link on the index under the first bullet point

File: docs/source/en/api/diffusion_pipeline.md (hunk 2 above)
Comment (stevhliu) [reply docs/source/en/api/diffusion_pipeline.md:1]: The diffusion pipeline is shown [here](http://localhost:3000/api/pipelines/overview#diffusers.DiffusionPipeline) with the pipeline overview similar to how we show the base scheduler/model classes on their overview pages.

File: docs/source/en/api/pipelines/overview.md (hunk 6 above)
Comment (stevhliu) [reply docs/source/en/api/pipelines/overview.md:46]: DeepFloyd IF should be on line 42 :)

File: docs/source/en/using-diffusers/custom_pipeline_examples.md [hunk 8]
```diff
@@ -14,273 +14,106 @@ specific language governing permissions and limitations under the License.
...
-If a community doesn't work as expected, please open an issue and ping the author on it.
+For more context about the design choices behind community pipelines, please have a look at [this issue](https://github.com/huggingface/diffusers/issues/841).
 
-| Example                                | Description                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              | Code Example                                                      | Colab                                                                                                                                                                                                              |                                                     Author |
```
[the remark this answers is not in the document]
Comment (stevhliu) [reply docs/source/en/using-diffusers/custom_pipeline_examples.md:23]: I think it's more lightweight to let the table [here](https://github.com/huggingface/diffusers/tree/main/examples/community#community-examples) provide the overview, and we can keep a link to it from the docs.

**Review by patrickvonplaten — APPROVED** [review]

File: docs/source/en/api/diffusion_pipeline.md [hunk 9]
```diff
@@ -1,36 +0,0 @@
...
-specific language governing permissions and limitations under the License.
--->
-
-# Pipelines
-
-The [`DiffusionPipeline`] is the quickest way to load any pretrained diffusion pipeline from the [Hub](https://huggingface.co/models?library=diffusers) for inference.
-
-<Tip>
-
-You shouldn't use the [`DiffusionPipeline`] class for training or finetuning a diffusion model. Individual
```
Comment (patrickvonplaten) [inline docs/source/en/api/diffusion_pipeline.md:19]: Could it make sense to move this Tip over or the other Diffusion Pipeline doc? It's a bit different from what is there already, but might be useful!
